### PR TITLE
Enhance test methodology

### PR DESCRIPTION
**Pull request by guzzilar**
Changes:

- Support PHPUnit test with local test (don't need an internet for request to Omise server anymore) *(For more detail, I added mock data (json files in fixtures folder) for run PHPUnit test without connect Omise server (local test))*
- Changed execute method behaviour (/lib/omise/res/OmiseApiResource.php) for support local test mode 
- Support automate testing with CircleCI (added circle.yml & phpunit.xml)
- Fixed php syntax for support PHP v5.3.20

### CODE REVIEW

File: tests/omise/TestConfig.php [hunk 1]
```diff
@@ -1,4 +1,4 @@
 <?php
 
-define('OMISE_PUBLIC_KEY', 'pkey');
-define('OMISE_SECRET_KEY', 'skey');
+define('OMISE_PUBLIC_KEY', 'pkey_test_4zfuolwdegumf4k2qba');
+define('OMISE_SECRET_KEY', 'skey_test_4zfuom5ugbndd4z8nlf');
```
Comment (robinclart) [inline tests/omise/TestConfig.php]: Does this file still make sense since we're not really calling the server?

File: tests/omise/RefundTest.php [hunk 2]
```diff
@@ -75,7 +75,8 @@ public function testRetrieve() {
     $charge = OmiseCharge::retrieve(self::$_charge['id']);
     $refunds = $charge->refunds();
 
-  	$refund = $refunds->retrieve($refunds->create(array('amount' => 10000))['id']);
+    $create = $refunds->create(array('amount' => 10000));
+  	$refund = $refunds->retrieve($create['id']);
```
Comment (robinclart) [inline tests/omise/RefundTest.php]: There's some weird indentation in this file. Please align everything at the same level with the same indent.

Comment (guzzilar) [inline tests/omise/RefundTest.php]: oops sorry, my text editor convert a space to \tab. I'll fix it now

**Comment by robinclart** [comment]
Also in the README, the test section has not been updated to reflect the change in this PR.

**Comment by robinclart** [comment]
Actually in all files the indentation is a mix of tabs and spaces. Could you change everything to 2 spaces in all files and align things properly.

**Comment by robinclart** [comment]
Also. The file OmiseApiResourceSingleton is no longer used I think.

File: tests/omise/TestConfig.php (hunk 1 above)
Comment (guzzilar) [inline tests/omise/TestConfig.php]: Ok, I tried to remove this file from test case file and run test again, It's fine.
I think it not necessary for now (but not anymore, I kept it for test with server sometimes).

I'll clean my pkey, skey from this file to `pkey`, `skey` (default) but I think it should keep this file in tests folder not remove it.

Comment (robinclart) [inline tests/omise/TestConfig.php]: Ok

**Comment by guzzilar** [comment]
For file indent, I agree with you (but for your information, If we need to write a code follow a PSR standard, It should 4 spaces not 2 || ref: http://www.php-fig.org/psr/psr-2/)

For OmiseApiResourceSingleton.php file.
Actually, I have no idea (I'm not sure we really don't need it anymore in the future) But I'm ok for delete it.

**Comment by robinclart** [comment]
Up to you. If you think PSR-2 is good then go ahead and change all files to conform to it. And add it to the README for people who want to contribute (maybe not in this branch).

**Comment by robinclart** [comment]
For OmiseApiResourceSingleton.php if we need it later we can bring it back. This is git after all. Don't be attach to code.

**Comment by robinclart** [comment]
Good job :+1:

**Pull request merged**